tests/char_suffer_test.cpp

```diff
@@ -0,0 +1,335 @@
+#include "catch/catch.hpp"
+
+#include "avatar.h"
+#include "game.h"
+#include "map.h"
+#include "map_helpers.h"
+#include "player_helpers.h"
+
+// Tests for Character suffering
+//
+// Covers functions:
+// - Character::suffer_from_sunburn
+
+// Main body parts that can have HP loss
+const std::vector<bodypart_str_id> body_parts_with_hp{
+    bodypart_str_id( "head" ), bodypart_str_id( "torso" ),
```

Comment:
you could replace this with get_all_bodyparts( true )

Reply:
I did it this way here because I needed the name of the body part, which I could not figure out how to get from `bodypart_id` (which just uses integers). `Creature::get_all_body_parts( true )` returns `bodypart_id`, not `bodypart_str_id`.

Do you know of some way to get the part name from a `bodypart_id`?

Reply:
`int_id` offers access to the underlying part, just like `string_id` does.  So you can just do
```c++
bodypart_id bp = ...;
bp->name();
```
Also, you can convert between `int_id` and `string_id` using the `id()` member function; you're already doing this in the `suffer.cpp` code.

Reply:
@jbytheway Thanks - I tried `bp->name()` but it did not work for me (giving the below error from clang) but I got it working well with `bp.id().str()` so I went with that for now.

```
char_suffer_test.cpp:219:30: error: type 'const translation' does not provide a call operator
                    CAPTURE( bp->name() );
                             ^~~~~~~~
```

Reply:
Ad, so you wanted the id; I thought you meant the translated name.